From 11b6597b00a25313cc53d396e989daa5c3bd8703 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 06 May 2019 14:34:11 +0300
Subject: [PATCH] [SW-0] pll: configure all available divisors

Current code does not configure divisors which are specified as
HL_PLL_SKIP in the div_sel[] array.
To prepare for the IF/MESH W/A addition, configure all divisors,
yet keep the ability to skip the clock selector configuration.

Change-Id: I327206ab375d7a4951d63433a718825ce815450c
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index 5438faf..2c2409a 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -104,7 +104,7 @@
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
 		if (cfg->div_sel[sel] == HL_PLL_REF_CLK ||
-		    cfg->div_sel[sel] == HL_PLL_SKIP)
+		    cfg->div_factor[sel] < 0)
 			continue;
 
 		div_base = cfg->base + (sel * sizeof(u32));
diff --git a/common/pll.h b/common/pll.h
index 68bd340..e1ccf34 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -53,7 +53,7 @@
 	u32	nf;
 	u32	od;
 	u32	nb;
-	u32	div_factor[HL_PLL_DIV_SIZE];
+	int	div_factor[HL_PLL_DIV_SIZE];
 	int	div_sel[HL_PLL_DIV_SIZE];
 };
 
